### PR TITLE
Slurm failover with nfs shared directory

### PR DESCRIPTION
**Pull request by schlad**
- Related ticket: 
https://progress.opensuse.org/issues/40718
- Verification run

(1) new tests for CTL failover:
supportserver:
http://10.160.66.198/tests/4725
ctl main
http://10.160.66.198/tests/4726
ctl backup:
http://10.160.66.198/tests/4727
compute nodes:
http://10.160.66.198/tests/4728
http://10.160.66.198/tests/4729

(2) the test does use the same barriers so here is the run demonstrating nothing is broken:
supportserver:
http://10.160.66.198/tests/4720
ctl:
http://10.160.66.198/tests/4724
compute nodes:
http://10.160.66.198/tests/4722
http://10.160.66.198/tests/4723
http://10.160.66.198/tests/4724

### CODE REVIEW

File: tests/hpc/slurm_master_adv.pm [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    assert_script_run("srun -N ${nodes} /bin/ls");
+    assert_script_run("sinfo -N -l");
+    assert_script_run("scontrol ping");
+
+    # backup ctl takes over
+    assert_script_run("scontrol takeover");
+    assert_script_run("scontrol ping");
+
+    assert_script_run("srun -w slurm-slave01 date");
+    assert_script_run("srun -w slurm-slave02 date");
```
Comment (czerw) [inline tests/hpc/slurm_master_adv.pm]: Would be possible to run commands by on all compute nodes? (according to number of compute nodes)

Comment (schlad) [reply tests/hpc/slurm_master_adv.pm]: @czerw sure. I did this on purpose  - check if srun can work with only one compute node. And there is: srun -N ${nodes} /bin/ls which does check if srun works on all compute nodes.
needless to say - I have an idea to further enhance "reusing of the created cluster" to run more jobs with more parameters and check if this works well.

File: tests/hpc/slurm_master_backup.pm [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    $self->enable_and_start('munge');
+
+    # enable and start slurmctld
+    $self->enable_and_start('slurmctld');
+    systemctl 'status slurmctld';
+
+    # enable and start slurmd since maester also acts as Node here
+    $self->enable_and_start('slurmd');
+    systemctl 'status slurmd';
+
```
Comment (czerw) [inline tests/hpc/slurm_master_backup.pm]: Looks like repeating pattern, maybe status check could be included in `enable_and_start` or new function could be created `enable_start_check`, but it is not important.

Comment (schlad) [reply tests/hpc/slurm_master_backup.pm]: @czerw yeah, fully agree. There is quite a few places which could go into certain functions. I will do that and perhaps eventually have two run() - init cluster and then another one for executing the test procedures itself. Further enhancements as you propose will be done with slurm with database test case. And then even further enhancements should be done once I look into other components of the hpc, so eventually hpclib should provide generic functions for preparing fully functional cluster and there should be no duplication of the code in the test procedures.

**Comment by asmorodskyi** [comment]
cool !

**Pull request merged**